### PR TITLE
feat(page): check activation account 

### LINKED ISSUE #28: Check activation account
## Description

Check account activation

### Figma Links

- [Mobile](https://www.figma.com/file/a4l6zqfnLyjG9h9Qyj6QsE/Silacak?node-id=189%3A2074)
- [Desktop](https://www.figma.com/file/a4l6zqfnLyjG9h9Qyj6QsE/Silacak?node-id=143%3A2810)
- [Prototype](https://www.figma.com/proto/a4l6zqfnLyjG9h9Qyj6QsE/Silacak?node-id=143%3A2810&scaling=scale-down-width&page-id=0%3A1&starting-point-node-id=4%3A2&hide-ui=1)

Todo:

1. [ ] Add button in login page to redirect to check activation account page
2. [ ] Form title
3. [ ] Form text input
4. [ ] Form button check
5. [ ] Message Error user is not activated
6. [ ] Message Error user not found
7. [ ] Message Success user is activated

### PR DESCRIPTION
**Pull request by asaadam**
Closes #28 
## Current Tasks

Todo:

- [ ]  Add button in login page to redirect to check activation account page
- [X]  Form title
- [X]  Form text input
- [X]  Form button check
- [X]  Message Error user is not activated
- [X]  Message Error user not found
- [X]  Message Success user is activated

### CODE REVIEW

**Review by resir014 — CHANGES_REQUESTED** [review]

**Comment by resir014** [comment]
I think it's about time we start implementing a form library before it's too late. What do you think, @zainfathoni?

For ease of use, we can use one of these two form libraries:

- [Formik](https://formik.org/) ([bundlephobia](https://bundlephobia.com/package/formik@2.2.9))
- [react-hook-form](https://react-hook-form.com/) ([bundlephobia](https://bundlephobia.com/package/react-hook-form@7.12.2))

File: components/layout/checkAccount/intex.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from "./check-account-form";
```
Comment (akrifari) [inline components/layout/checkAccount/intex.ts:1]: Did you mean `components/layout/checkAccount/index.ts`?

Comment (asaadam) [reply components/layout/checkAccount/intex.ts:1]: ah ya, thanks for the correction

**Comment by asaadam** [comment]
also, I want to ask, 
about fetch lib, do we need to implement fetch library? 
like axios or react-query?

**Comment by zainfathoni** [comment]
> I think it's about time we start implementing a form library before it's too late. What do you think, @zainfathoni?
> 
> For ease of use, we can use one of these two form libraries:
> 
> * [Formik](https://formik.org/) ([bundlephobia](https://bundlephobia.com/package/formik@2.2.9))
> * [react-hook-form](https://react-hook-form.com/) ([bundlephobia](https://bundlephobia.com/package/react-hook-form@7.12.2))

Agreed, let's use `react-hook-form`, since it's more intuitive for us to use hooks, and we can also extract the hooks into custom hooks easily.
Moreover, it promises a good performance with a very small bundle size. We should always optimize for performance in this project since one of the most significant users' complaints was the performance of the app.

**Comment by zainfathoni** [comment]
> also, I want to ask,
> about fetch lib, do we need to implement fetch library?
> like axios or react-query?

Please just use `fetch`, so we don't add unnecessary overhead to the app. Again, we should always optimize for performance in this project since one of the most significant users' complaints was the performance of the app.

You can read more about it in [our principles](https://kcov.id/wbw-principles).

**Comment by resir014** [comment]
@asaadam @zainfathoni If we still want to add something to make fetching easier, we can also use [swr](https://swr.vercel.app/) which is a more lightweight version of react-query.

**Comment by zainfathoni** [comment]
> @asaadam @zainfathoni If we still want to add hooks to make fetching easier, we can also use [swr](https://swr.vercel.app/) which is a more lightweight version of react-query.
> 
> * [Bundle stats for react-query](https://bundlephobia.com/package/react-query@3.19.6)
> * [Bundle stats for swr](https://bundlephobia.com/package/swr@0.5.6)
> 
> But if it's possible, please still use regular `fetch`.

That's a good idea. I think `fetch` would be too verbose since we will be performing a lot of API requests on the client-side, unlike our wargabantuwarga.com website where almost no API calls are made on the client-side.
Therefore, let's use `swr` to perform our API interactions.

**Comment by asaadam** [comment]
hello mas @zainfathoni and mas @resir014 

image

should I declare this orange color on `tailwind.config.js` ?

**Comment by zainfathoni** [comment]
What do you think, @resir014?

In my opinion, we should simply use the existing color palette to reduce the cognitive load of picking which colors to use. And since we don't need to achieve a pixel-perfect standard for this iteration, I think a little bit of color deviation should be tolerated as well.

**Comment by asaadam** [comment]
hello, again mas @zainfathoni and mas @resir014 could you check my implementation on SWR and react-hook-form? 

I think it gonna be a lot of corrections 😅 because it's my first time using that lib.

**Comment by asaadam** [comment]
also, I got cors response, should I create reserve-proxy?

**Comment by asaadam** [comment]
as per @ramarahmanda says, I remove SWR implementation. 
and just finish layouting first

**Comment by zainfathoni** [comment]
Please test your new files so that it passes the codecov CI checks, @asaadam. Thanks! 🙏

**Comment by asaadam** [comment]
> Please test your new files so that it passes the codecov CI checks, @asaadam. Thanks! 🙏

done yak mas, please check it again

**Review by resir014 — APPROVED** [review]

**Comment by zainfathoni** [comment]
@all-contributors please add @asaadam for testing

**Pull request merged**